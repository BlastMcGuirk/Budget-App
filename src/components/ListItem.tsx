import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { FontSizes, Layouts } from '../styles/global';

export interface Item {
    name: string;
    amount: number;
    date: string;
    category?: string;
}

export interface ListItemProps {
    item: Item
}

export default function ListItem(props: ListItemProps) {
    const { name, amount, date, category } = props.item;
    return (
        <View style={styles.container}>
            <View style={Layouts.row}>
                <Text style={FontSizes.L}>{name}</Text>
                <Text style={FontSizes.L}>{amount}</Text>
            </View>
            <View style={Layouts.row}>
                <Text style={FontSizes.S}>{category}</Text>
                <Text style={FontSizes.S}>{date}</Text>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        padding: 16,
        borderBottomColor: 'gray',
        borderBottomWidth: 1
    },
})