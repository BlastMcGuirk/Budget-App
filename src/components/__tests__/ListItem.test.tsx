import renderer from 'react-test-renderer';
import ListItem, { Item } from '../ListItem';

describe("List Item Tests", () => {

    const itemWithCategory: Item = {
        name: "Item",
        amount: 12.34,
        date: "1/1/22",
        category: "Category"
    }
    
    const itemWithoutCategory: Item = {
        name: "Item",
        amount: 12.34,
        date: "1/1/22",
    }

    it("Renders correctly with category", () => {
        const tree = renderer
            .create(<ListItem item={itemWithCategory}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Renders correctly with category", () => {
        const tree = renderer
            .create(<ListItem item={itemWithoutCategory}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });


});