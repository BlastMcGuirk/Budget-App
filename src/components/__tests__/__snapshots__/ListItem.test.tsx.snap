// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item Tests Renders correctly with category 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "gray",
      "borderBottomWidth": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Item
    </Text>
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      12.34
    </Text>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Category
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      1/1/22
    </Text>
  </View>
</View>
`;

exports[`List Item Tests Renders correctly with category 2`] = `
<View
  style={
    Object {
      "borderBottomColor": "gray",
      "borderBottomWidth": 1,
      "padding": 16,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Item
    </Text>
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      12.34
    </Text>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      1/1/22
    </Text>
  </View>
</View>
`;
